contracts/adapters/AaveMigrationExtension.sol

```diff
@@ -0,0 +1,578 @@
+     * @param premium The additional fee charged for the flash loan.
+     * @param initiator The initiator of the flash loan.
+     * @param params Encoded migration parameters.
+     * @return True if the operation is successful.
+     */
+    function executeOperation(
+        address asset,
+        uint256 amount,
+        uint256 premium,
+        address initiator,
+        bytes calldata params
+    )
+        external
+        override
+        returns (bool) 
+    {
+        require(msg.sender == address(POOL), "MigrationExtension: invalid flashloan sender");
+        require(asset == address(underlyingToken), "MigrationExtension: invalid flashloan asset");
+        require(initiator == address(this), "MigrationExtension: invalid flashloan initiator");
+
+        // Decode parameters and migrate
+        DecodedParams memory decodedParams = abi.decode(params, (DecodedParams));
+        _migrate(decodedParams);
+
+        // Approve flashloan repayment
+        uint256 totalAmount = amount + premium;
+        underlyingToken.approve(address(POOL), totalAmount);
+        return true;
+    }
+
+    /**
+     * @notice Receives ERC721 tokens, required for Uniswap V3 LP NFT handling.
+     * @dev Callback function for ERC721 token transfers, enabling the contract to receive Uniswap V3 LP NFTs. Always returns the selector to indicate successful receipt.
+     * @return The selector of the `onERC721Received` function.
+     */
+    function onERC721Received(
+        address, // operator
+        address, // from
+        uint256, // tokenId
+        bytes calldata // data
+    ) 
+        external
+        override
+        returns (bytes4)
+    {
+        return this.onERC721Received.selector;
+    }
+
+    /**
+     * @notice OPERATOR ONLY: Transfers any residual balances of the underlying token, Aave's 
+     * wrapped token, and the wrapped SetToken to the operator's address.
+     * @dev This function is intended to recover tokens that might have been left behind
+     * due to the migration process or any other operation. It ensures that the contract
+     * does not retain any assets inadvertently. Only callable by the operator.
+     */
+    function sweepTokens() external onlyOperator {
+        // Sweep underlying token
+        uint256 underlyingTokenBalance = underlyingToken.balanceOf(address(this));
+        if (underlyingTokenBalance > 0) {
+            underlyingToken.transfer(manager.operator(), underlyingTokenBalance);
+        }
+
+        // Sweep Aave's wrapped token
+        uint256 aaveTokenBalance = aaveToken.balanceOf(address(this));
+        if (aaveTokenBalance > 0) {
+            aaveToken.transfer(manager.operator(), aaveTokenBalance);
+        }
+
+        // Sweep wrapped SetToken
+        uint256 wrappedSetTokenBalance = wrappedSetToken.balanceOf(address(this));
+        if (wrappedSetTokenBalance > 0) {
+            wrappedSetToken.transfer(manager.operator(), wrappedSetTokenBalance);
+        }
+    }
+
```

Comment:
We might to add a operator-only method to make arbitrary calls on behalf of this contract, to be able to recover from any unforseen contract / migration state.  Not sure if necessary / advisable from security standpoint.

Reply:
Considering unnecessary because the contract already has separate endpoints for 
+ Trade Module interactions
+ Uniswap V3 liquidity interactions
+ Collateral token sweeps